content/blog/the-importance-of-ubiquitous-language/index.md

```diff
@@ -0,0 +1,188 @@
+
+How can a "ubiquitous" language help how you talk to your teams, your users, and the business (stakeholder and end-users)?  What do you need to do to discover it, and how should it be used to be most effective?  Before you dive into the answers to these questions, let's start with the basics.
+
+## What is ubiquitous language?
+Ubiquitous language is a set of unambiguous vocabulary shared by all members and stakeholders of a product team.  Eric Evans talks "Ubiquitous Language" in his book [Domain Driven Design: Tackling Complexity in the Heart of Software](https://www.amazon.com/gp/product/0321125215). 
+
+> Ubiquitous Language is modeled within a Limited context, where the terms and concepts of the business domain are identified, and there should be no ambiguity.
+
+For those working on a [balanced team](https://tanzu.vmware.com/developer/learningpaths/application-development/balanced-teams/), this means that the product manager(s), designer(s), and engineer(s) use the same terminology to talk about their product, but _**more importantly**_, the same language that is used to talk to the business.  What makes this language ubiquitous is that it is truly **found everywhere**, from communications with the end-users and the stories derived from those conversations, all the way down to the objects and code that is written as a result. According to Evans, the language should align with your [domain model](https://martinfowler.com/eaaCatalog/domainModel.html), the structure of the interconnected objects and their associated methods defining a business process, of the software itself.  
+
+Software code, by necessity, is very structured and unambiguous.  For example, the same word in the same context in a block of code cannot have two different meanings because the compiler would be unable to determine how to compile the code to be understood by the computer.  However, language (especially English) can be highly ambiguous.  Take the following sentence for example.  
+
+> The ball would have hit the bat if it were two inches higher.
+
+The word "bat" in this sentence could have referred to a bat with wings or a bat made of wood.  The sentence is correct either way, and absent any additional context, you do not know the true meaning.
+
+By maintaining a one-to-one correlation of the language you use to describe the software you are building, with the actual software, you can utilize the un-ambiguous nature of the code to drive more ambiguity out of the language you use to describe it.  If you are able to create a way of communicating that is not ambiguous, your process of defining and building software can become a lot more efficient.  This can be accomplished by removing the need to ask questions like "what do you mean by that?", or "do you mean..." as frequently.  It also spares you the embarrassment of getting all the way from a user interview, to a user story, to code and then realizing, "Oh!  You meant this...".
```

Comment:
```suggestion
By maintaining a one-to-one correlation of the language you use to describe the software you are building, with the actual software, you can utilize the unambiguous nature of the code to drive more ambiguity out of the language you use to describe it.  If you are able to create a way of communicating that is not ambiguous, your process of defining and building software can become a lot more efficient.  This can be accomplished by removing the need to ask questions like "what do you mean by that?", or "do you mean..." as frequently.  It also spares you the embarrassment of getting all the way from a user interview, to a user story, to code and then realizing, "Oh!  You meant this...".
```